File: Pi-Javafx-Symfony-yahya/src/main/java/tn/esprit/services/StripeService.java
package tn.esprit.services;

import com.stripe.exception.StripeException;
import com.stripe.model.PaymentIntent;
import java.util.HashMap;
import java.util.Map;

public class StripeService {

    public static PaymentIntent createPayment(double amount, String currency) {
        try {
            Map<String, Object> params = new HashMap<>();
            params.put("amount", (int) (amount * 100)); // Stripe requires the amount in cents
            params.put("currency", currency);
            params.put("payment_method_types", new String[]{"card"}); // Card payment

            return PaymentIntent.create(params);
        } catch (StripeException e) {
            e.printStackTrace();
            return null;
        }
    }
}

File: src/Service/StripeService.php
<?php

namespace App\Service;

use Stripe\Stripe;
use Stripe\PaymentIntent;

class StripeService
{
    public function __construct()
    {
        // 🔥 API KEY DIRECTEMENT ICI
        Stripe::setApiKey('sk_test_51QzP2uPFHNYg7RkVtRbawzVoW57klCf9OyWZ7r9zzUV7X9F8exNoDMFxUyiWqpVFr7jvGapsKGRsuBGFtLSvM34l00kPHtYFrm');
    }

    public function createPaymentIntent(float $amount, string $currency = 'usd'): ?PaymentIntent
    {
        try {
            return PaymentIntent::create([
                'amount' => (int)($amount * 100), // Stripe attend des CENTS
                'currency' => $currency,
                'payment_method_types' => ['card'],
            ]);
        } catch (\Exception $e) {
            return null;
        }
    }
}
